### PR TITLE
fix(administration) : Make the users not created on centreon plateform unenabled to edit their profile information

### PR DESCRIPTION
**Pull request by Exodiaaaa**
## Description

Users created not using Centreon Plateform cannot update their information(Alias, Name, Email, Password Management)

**Fixes** # (MON-24798)

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [X] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Given a user authenticated through SSO (or LDAP) and having rights to access to “Administration > My Account”
When he access to “Administration > My Account” menu
He can’t change his following properties:

- Name
- Alias / Login
- Email

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by arnapou — APPROVED** [review]

**Comment by jeremyjaouen** [comment]
There is an underlying error on authentication.
I have dump the contact auth type on the formMyAccount file
image
image

It appears that my auth type is "local", but i'm authenticate trough OpenId. So i'm still able to modify my informations, then  broke my authentication.

image
image


image

**Review by arnapou — APPROVED** [review]

File: centreon/www/include/Administration/myAccount/formMyAccount.php [hunk 1]
```diff
@@ -418,7 +422,7 @@ function myReplace()
 $sessionKeyFreeze = 'administration-form-my-account-freeze';
 
 if ($form->validate()) {
-    updateContactInDB($centreon->user->get_id());
+    updateContactInDBIfLocal($cct['contact_id']);
```
Comment (callapa) [inline centreon/www/include/Administration/myAccount/formMyAccount.php:425]: The method name suggests that the update is only performed if the user is 'local', which is not true.

Comment (Exodiaaaa) [reply centreon/www/include/Administration/myAccount/formMyAccount.php:425]: i suggest those name for the method : progressiveContactUpdate - contextualContactUpdate -incrementalContactUpdate 
Which one do you think is the best ?

File: centreon/www/include/Administration/myAccount/DB-Func.php [hunk 2]
```diff
@@ -258,3 +258,77 @@ function checkAutologinValue(array $fields)
...
+            ':contactTheme',
+            !empty($ret['contact_theme']['contact_theme']) ? $ret['contact_theme']['contact_theme'] : "light",
+            \PDO::PARAM_STR
+        );
+        $stmt->bindValue(':defaultPage', !empty($ret['default_page']) ? $ret['default_page'] : null, \PDO::PARAM_INT);
+        $stmt->bindValue(':showDeprecatedPages', isset($ret['show_deprecated_pages']) ? 1 : 0, \PDO::PARAM_STR);
+        $stmt->bindValue(':contactId', $contact_id, \PDO::PARAM_INT);
+        $stmt->execute();
+        $stmt->closeCursor();
+        if (isset($ret["contact_passwd"]) && !empty($ret["contact_passwd"])) {
```
Comment (arnapou) [inline centreon/www/include/Administration/myAccount/DB-Func.php:318]: `empty` is already doing an `isset` check internally.
```suggestion
        if (!empty($ret["contact_passwd"])) {
```

File: centreon/www/include/Administration/myAccount/DB-Func.php [hunk 3]
```diff
@@ -258,3 +258,77 @@ function checkAutologinValue(array $fields)
 
     return count($errors) > 0 ? $errors : true;
 }
+function progressiveContactUpdate($contact_id = null){
```
Comment (callapa) [inline centreon/www/include/Administration/myAccount/DB-Func.php:261]: Why progressive ?

**Comment by Exodiaaaa** [comment]
[tests.zip](https://github.com/centreon/centreon/files/13599068/tests.zip)
this is functional tests for my pr

**Pull request closed without merging**